DF-2162 - Adds Working at the CFPB  page

anselmbradford opened the pull request:
Adds initial careers - working at the CFPB page.

## Additions

- Added `/careers/working-at-cfpb/` page. 
- Added block templates for current openings, job app process, linkedin info, and provide feedback link.
- Added `.content-l_col__before-inset-divider` CF enhancement for a column divider in cf-layout that has a gap between the top of the divider and the top border. 
- Added `MobileCarousel.js` module for instantiating the slick carousel and added associated `js-mobile-carousel` class as a hook.

## Changes

- Moved modules that can be instantiated through the `new` keyword to a `classes` subdirectory.
- Moved page-sniffing JS code to page scripts for the-bureau and working with the CFPB pages.

## Testing

- `careers/working-at-cfpb/` should work.
- the-bureau page carousels should still function.

## Review

- @KimberlyMunoz 
- @jimmynotjim 
- @sebworks 

## Preview

Desktop:
![screen shot 2015-06-29 at 10 53 48 am](https://cloud.githubusercontent.com/assets/704760/8410610/5d5d7ba4-1e4d-11e5-85c8-f3c64363a48b.png)
![screen shot 2015-06-29 at 10 54 01 am](https://cloud.githubusercontent.com/assets/704760/8410615/62cbe328-1e4d-11e5-8b8f-1e722977d594.png)
![screen shot 2015-06-29 at 10 54 13 am](https://cloud.githubusercontent.com/assets/704760/8410617/66d6f278-1e4d-11e5-92c7-9adaf51fa48e.png)

Mobile carousel:
![screen shot 2015-06-29 at 10 54 37 am](https://cloud.githubusercontent.com/assets/704760/8410625/761a2610-1e4d-11e5-8eae-70bbefcaeee2.png)

## Notes

- Bottom-right border of the footer area with be enabled with https://github.com/cfpb/cfgov-refresh/pull/684
- summaries block templates are in the global includes, but some of them (current jobs, job app process) likely will only be in the careers section. We may want to establish a pattern for section-specific templates. Events has some, but given the evolution of the project they could use some organization.

sebworks commented:
Is the spacing correct on mobile?

![screen shot 2015-06-29 at 11 23 19 am](https://cloud.githubusercontent.com/assets/1696212/8411315/65cae822-1e51-11e5-9ba4-821ba6ecdfd0.png)

sebworks commented on careers/working-at-cfpb/index.html:
```diff
@@ -0,0 +1,184 @@
+            </p>
+    </section>
+
+    <div>
```
Are these extra divs necessary?

anselmbradford replied:
I could of sworn I had them there for a reason, but I can't seem to find out why. Removed and the layout looks the same in https://github.com/cfpb/cfgov-refresh/commit/295f54d5581add948b58a701ca63ae0f6f8e2bce

KimberlyMunoz commented on _includes/templates/summaries/linkedin.html:
```diff
@@ -0,0 +1,16 @@
+<h2 class="h4">
+    <span class="cf-icon cf-icon-linkedin-square"></span>
+    Follow us on LinkedIn
```
We're using title case everywhere now 🎊

KimberlyMunoz replied:
Actually, this is fine.

jimmynotjim replied:
"Us" should still be title case though right?

anselmbradford replied:
:+1:

anselmbradford replied:
@sarahsimpson09 Should it be `Follow Us On LinkedIn`?

jimmynotjim replied:
Title case typically does not capitalize the first letter of "on", "in", "the", "and", "of", "or", or "at"

anselmbradford replied:
Updated to `Follow Us on Linkedin` in https://github.com/cfpb/cfgov-refresh/commit/ed4e0c971c8550360bf26cdfd3d295f7e5ce4ff7

sarahsimpson09 replied:
Yes--should say "Follow us on LinkedIn"

On Tue, Jun 30, 2015 at 3:39 PM, Ans <notifications@github.com> wrote:

> In _includes/templates/summaries/linkedin.html
> <https://github.com/cfpb/cfgov-refresh/pull/686#discussion_r33614544>:
>
> > @@ -0,0 +1,16 @@
> > +<h2 class="h4">
> > +    <span class="cf-icon cf-icon-linkedin-square"></span>
> > +    Follow us on LinkedIn
>
> @sarahsimpson09 <https://github.com/sarahsimpson09> Should it be Follow
> Us On LinkedIn?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfpb/cfgov-refresh/pull/686/files#r33614544>.
>

KimberlyMunoz commented on careers/working-at-cfpb/index.html:
```diff
@@ -0,0 +1,184 @@
+                        [ '/newsroom/',
+                          'Newsroom' ],
+                        [ '',
+                          'Office of Civil Rights' ]
```
The mockups have Office of Civil Rights, but the final copy has Equal Employment Opportunity. This is on all the Career pages, so we should get final word from @duelj and @sarahsimpson09

anselmbradford replied:
Office of Civil Rights it is https://github.com/cfpb/cfgov-refresh/pull/686#issuecomment-116754972

sebworks commented:
Carousel doesn't appear to be centered at some points.

sarahsimpson09 commented:
Yes, we're changing EEO to say Office of Civil Rights more broadly. Thanks
for double checking!

On Mon, Jun 29, 2015 at 12:38 PM, sebworks <notifications@github.com> wrote:

> Carousel doesn't appear to be centered at some points.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfpb/cfgov-refresh/pull/686#issuecomment-116753926>.
>

anselmbradford commented:
Pushed a big update in https://github.com/cfpb/cfgov-refresh/commit/b9c85b3ada628b4972263152a26c7c38ccba7a45:
- Moves the carousel to a macro, so that working at the CFPB and the bureau now share the same carousel component.
- Added `the-bureau` page class to the bureau.
- Fixed code comments in /macros subdirectory.

anselmbradford commented:
Working at the CFPB carousel
![screen shot 2015-06-29 at 3 07 06 pm](https://cloud.githubusercontent.com/assets/704760/8416968/59b6efd8-1e77-11e5-90c9-f5404d98caaf.png)

The bureau carousel 
![screen shot 2015-06-29 at 3 33 02 pm](https://cloud.githubusercontent.com/assets/704760/8416956/49c0b898-1e77-11e5-9154-441dcb2f48b0.png)

sebworks commented:
:+1:

jimmynotjim commented:
Should the dividers and the content footer align? 

![screen shot 2015-06-29 at 5 58 28 pm](https://cloud.githubusercontent.com/assets/1280430/8419539/81bc0b10-1e88-11e5-8e56-c340cafea668.png)

jimmynotjim commented:
Should the background image center as you change screen size?

![screen shot 2015-06-29 at 6 00 41 pm](https://cloud.githubusercontent.com/assets/1280430/8419572/daa50e20-1e88-11e5-9ae1-3c67510025d8.png)

jimmynotjim commented:
Should these align on the left? 

![screen shot 2015-06-29 at 6 02 12 pm](https://cloud.githubusercontent.com/assets/1280430/8419596/02c1739e-1e89-11e5-858f-3ca60a2a982f.png)

jimmynotjim commented:
Looks like `.content-l_col__before-inset-divider.content-l_col-1-2` has a border left of 30px all the time.

jimmynotjim commented on _includes/macros/mobile-carousel.html:
```diff
@@ -0,0 +1,50 @@
+   slides: An array of data to populate the carousel slides with.
+           The structure looks like:
+           {
+           'image_sizes_class': <A image sizing class for responsive sizes>,
```
L:16-25 should be indented once more.

anselmbradford replied:
Fixed in https://github.com/cfpb/cfgov-refresh/commit/103ef847f88472457936689fc4013cfaef2ab7f8

jimmynotjim commented on _includes/macros/mobile-carousel.html:
```diff
@@ -0,0 +1,50 @@
+
+   ========================================================================== #}
+
+{% macro render( slides ) %}
```
Spaces aren't used elsewhere within parens in the jinja files, this should probably be standardized.

anselmbradford replied:
It would be good idea to standardize this across all project filetypes (JS, CSS, Jinja). What's the preference?

anselmbradford replied:
Removed parens in https://github.com/cfpb/cfgov-refresh/commit/103ef847f88472457936689fc4013cfaef2ab7f8 till we work this out.

jimmynotjim replied:
My preference is for spaces, since that's the only standard we've had (in JS). But since it's not the way CSS or Jinja files currently are, go without for the time being.

jimmynotjim commented on _includes/templates/summaries/linkedin.html:
```diff
@@ -0,0 +1,16 @@
+    Connect with us to stay updated on the work
+    we do and new opportunities to be a part of it.
+</p>
+<p class="short-desc">
```
This isn't a description, should this be something else? Maybe just a link on it's own?

anselmbradford replied:
Fixed in https://github.com/cfpb/cfgov-refresh/commit/9c8ea4dd37896e21eeef696cc7c492ce98e7921c  and https://github.com/cfpb/cfgov-refresh/commit/fe406bb5534f5de096b88ca71e04f7729d656452

Are there any issues with an inline block (link) next to a block (paragraph)? I always feel like blocks should be sibling to blocks.

jimmynotjim replied:
Don't we have a modifier to make links block?

anselmbradford replied:
I'm not sure, @Scotchester ?

Scotchester replied:
There's `.block-link`, but it's mainly for adding padding and borders at small screen sizes.

I don't see a problem with a block having an inline-block sibling, if it renders fine. If you were to just apply `display: block` to it, the bottom border would stretch to the full width of the column.

jimmynotjim commented on _includes/templates/summaries/provide-feedback.html:
```diff
@@ -0,0 +1,18 @@
+        <li class="list_item">
+            <span class="cf-icon cf-icon-email list_icon"></span>
+            <span class="h5 list_text">Email</span>
+            <p><a href="mailto:jobs@consumerfinance.gov"
```
Can't nest a `<p>` inside another `<p>`

anselmbradford replied:
I just did it... mwhahaha :smiling_imp:  

… okay will fix

anselmbradford replied:
Fixed in https://github.com/cfpb/cfgov-refresh/commit/fe406bb5534f5de096b88ca71e04f7729d656452

jimmynotjim commented on _includes/templates/summaries/provide-feedback.html:
```diff
@@ -0,0 +1,18 @@
+<p class="short-desc">
+    We take pride in our feedback-centered culture
+    that helps us recruit great people.
+    If you have any feedback, or any specific questions, let us know at.
```
Should probably just end with "let us know", the "at" feels really odd.

anselmbradford replied:
I agree. Maybe I misheard @sarahsimpson09 .. can we take @jimmynotjim suggestion?

KimberlyMunoz replied:
I brought this up at team review, and the "at" is correct, but I think we should take Jimmy's suggestion.

jimmynotjim commented on careers/working-at-cfpb/index.html:
```diff
@@ -0,0 +1,150 @@
+        <div class="hero_img-container hero_img-container__small">
+            {# TODO: Replace placeholder div with image #}
+            <div class="hero_img"
+                 style="width:660px;height:300px;background-image:url({{ 'http://placehold.it/660x300.jpg' }});">
```
Should we set the width and height in the stylesheets? The image is required because we can't pass it through, but there doesn't seem to be a reason the dimensions need to be inline.

anselmbradford replied:
I'm not sure I understand. The whole div is a placeholder and will need to be replaced with an img when the actual image arrives. Unless the replacement image will be a background on a div too? The inline width/height are just because it won't pick up the width/height from the background image dimensions.

jimmynotjim replied:
Oh oh, sorry, I should have read the comment above it. Carry on :smile:

jimmynotjim commented on careers/working-at-cfpb/index.html:
```diff
@@ -0,0 +1,150 @@
+
+    <div class="block
+                block__flush-top">
+        <p class="short-desc">As a bureau, we support these values by focusing on equal opportunity, diversity, and inclusion. We value candidates from all backgrounds, traditional and non-traditional. We believe in investing in the growth of our colleagues and in creating an organization that is accountable to and representative of the American people.</p>
```
These are no longer descriptions. Maybe we should pull this style for wider re-use or rename the class something more descriptive?

jimmynotjim replied:
Add semantic line breaks here and on L:110

anselmbradford replied:
> These are no longer descriptions. Maybe we should pull this style for wider re-use or rename the class something more descriptive?

Yeah I'm not sure what to do here. This appeared to be the style needed to match the mocks. Does anyone know of a different one that fits here? Needs to stylistically look like:

![screen shot 2015-06-30 at 10 49 52 am](https://cloud.githubusercontent.com/assets/704760/8433768/ca66563a-1f15-11e5-93f5-77c154dc41b1.png)

cc @Scotchester @sebworks @KimberlyMunoz 

> Add semantic line breaks here and on L:110

How about this? https://github.com/cfpb/cfgov-refresh/commit/8bca73f22280940c715d46f5787eb04a9c6b1ec3

jimmynotjim replied:
Maybe we just need to change to an `.alt-font` class name for this style since it's being used outside of descriptions?

Scotchester replied:
This needs to move into its own issue. The problem is this: We want paragraphs of body copy of substantial length—length of the whole section of copy, not length of an individual paragraph—to be set in Georgia. Pretty much all other paragraphs are fair game for Avenir. Back when we started CF, we thought that paragraphs of body copy were the "default" case, and as such, set that as the base font family, overriding with Avenir Next when necessary. This led to the case of the not-very-semantic class `.short-desc`, among other unfortunate things like `.styled-link` and `.micro-copy`. (See: [cf-typography docs](https://cfpb.github.io/cf-typography/docs/).)

Since then, we've definitely had second thoughts about reversing our position on what the base font family should be. What do you all think about changing the base font family to Avenir Next (i.e., the sans-serif stack), and creating a new rule that sets "long-read" body copy in Georgia? (serif stack).

jimmynotjim replied:
I'm actually in favor of setting the reading text as the default (so keeping Georgia) and changing to the alt as necessary. I just think we need a better naming for it so it's clear what's being changed and why.

Scotchester replied:
Hey look! Here's the cf-core issue that already exists: cfpb/cf-core#22

anselmbradford replied:
So it sounds like for this PR this is okay, but needs to be addressed at a higher level.

Scotchester replied:
I concur.

jimmynotjim replied:
Yep.

jimmynotjim commented on careers/working-at-cfpb/index.html:
```diff
@@ -0,0 +1,150 @@
+                block__flush-top">
+        <p class="short-desc">As a bureau, we support these values by focusing on equal opportunity, diversity, and inclusion. We value candidates from all backgrounds, traditional and non-traditional. We believe in investing in the growth of our colleagues and in creating an organization that is accountable to and representative of the American people.</p>
+
+        <p class="short-desc">Our service to the people who work here begins with recruitment. It also includes professional development, competitive compensation and benefits, and financial wellness programs. We want our people to learn, improve, and push themselves to be great continuously.</p>
```
The last line feels like it's ending oddly. Could we update it to: "We want our people to continuously learn, improve, and push themselves."

anselmbradford replied:
What do you think @sarahsimpson09 ?

sarahsimpson09 replied:
I also thought this was weird. I had more of an issue with where "continuously" was in the sentence originally, but I like this change. I think go with "We are focused on continuously improving, learning, and pushing ourselves to be great"

anselmbradford replied:
Fixed in https://github.com/cfpb/cfgov-refresh/commit/2d378a4d186157ec31c59f42a544d2409c787281

jimmynotjim commented on src/static/css/mobile-carousel.less:
```diff
@@ -0,0 +1,113 @@
+   Overrides for slick carousel on /the-bureau/ and /careers/working-at-cfpb/.
+   ========================================================================== */
+
+.js-mobile-carousel {
```
`.js-` prefixed classes should be used for js hooks only. When styling them, just use the module name.

anselmbradford replied:
Which module name? Like add `mobile-carousel` class to the HTML and style it here?

jimmynotjim replied:
Exactly. None of the child blocks need the `.js-` prefix, since the scripting isn't targeting them specifically.

anselmbradford replied:
> None of the child blocks need the .js- prefix, since the scripting isn't targeting them specifically.

That's not entirely true. The Slick carousel attaches the `.slick-*` classes to the element with `.js-mobile-carousel` dynamically using JS, but I gotcha all the inside classes shouldn't have `js-` prefixes, such as `js-mobile-carousel_section`.

anselmbradford replied:
Fixed in https://github.com/cfpb/cfgov-refresh/commit/d889d89bc1b3711c4234fb5929d913b513cbf980

jimmynotjim replied:
You still don't want to use the `.js-` prefix for the styles, the parent block should have both `mobile-carousel` and `js-mobile-carousel` classes. `.js-` prefixes should only be used for JS hooks, not for styling. The entire idea around using the prefix is to decouple scripting from styling https://coderwall.com/p/qktuzw/decouple-javascript-classes-from-css-ones-by-using-prefix

anselmbradford replied:
Fixed in https://github.com/cfpb/cfgov-refresh/commit/88473ad0d5c44bf50e979695825f8c06779d3ac8

jimmynotjim commented on src/static/css/mobile-carousel.less:
```diff
@@ -0,0 +1,113 @@
+
+    // Slick carousel overrides https://github.com/kenwheeler/slick/.
+    &.slick-slider {
+          margin-top: (@grid_gutter-width * 2);
```
Remove extra indentation.

anselmbradford replied:
Fixed in https://github.com/cfpb/cfgov-refresh/commit/670bf6ef71c89891be9b2b3882a90a7cc40d219b

jimmynotjim commented on src/static/css/mobile-carousel.less:
```diff
@@ -0,0 +1,113 @@
+            }
+
+            li.slick-active {
+                button:before {
```
No need to nest since it's the only thing inside `li.slick-active`

anselmbradford replied:
Updated in https://github.com/cfpb/cfgov-refresh/commit/3506474f69b8b087c59a8f49cb7721760435f224 Is this what you meant? The button on L:54 doesn't need nesting, but L:69 does.

jimmynotjim replied:
Why does the button on L:69 need nesting? There's nothing else inside the `.slick-active` selector so you can just write it all as `.slick-active button:before {}`.

anselmbradford replied:
Gotcha, my mistake I misunderstood what you meant by nesting. Fixed in https://github.com/cfpb/cfgov-refresh/commit/761ecc8c125e9e3a944f4a8b6f0b10d034f2c28e

jimmynotjim commented on src/static/css/mobile-carousel.less:
```diff
@@ -0,0 +1,113 @@
+            }
+        }
+
+        .slick-prev,
```
Should always declare shared styles before individual styles (that way any individual over-rides are possible).

anselmbradford replied:
Fixed in https://github.com/cfpb/cfgov-refresh/commit/c73ba281a251339ec39e9ca3208fc2a52bb0230a

jimmynotjim commented on src/static/css/mobile-carousel.less:
```diff
@@ -0,0 +1,113 @@
+            height: 30px;
+            width: 30px;
+            top: 75px;
+            z-index: 1000;
```
Do we really need 1000?

anselmbradford replied:
Not sure. This was the existing Slick carousel override code. Slick needs to be updated anyway which may change this. I opened https://github.com/cfpb/cfgov-refresh/issues/692 for this issue.

jimmynotjim replied:
Ah, love when plugins overzealously set z-index. Why use 1 when 1000 is so much more awesome *sarcasm*

jimmynotjim commented on src/static/css/mobile-carousel.less:
```diff
@@ -0,0 +1,113 @@
+
+            &:before {
+                color: @gray-20;
+                opacity: 1;
```
Not sure if even a solid opacity causes the browser to hit the GPU, but using `opacity: initial` will probably be sure to prevent it.

anselmbradford replied:
Fixed in https://github.com/cfpb/cfgov-refresh/commit/bb16e25aa5543b1df99a8bd749d9b74f1a0793e3

jimmynotjim commented on src/static/css/pages/bureau.less:
```diff
@@ -6,33 +6,15 @@
+// Override general slick carousel overrides defined in mobile-carousel.less.
+// Removes bottom border, because the-bureau has a bordered block below
+// its carousel.
+.the-bureau {
```
Not necessary to nest any of these

anselmbradford replied:
We don't have per-page CSS so the `.the-bureau` scoping is necessary? The `.js-mobile-carousel` is not strictly necessary, but I'd argue this provides better encapsulation. If for some reason another Slick carousel was used on the page the styles wouldn't collide. Also, this follows the structure of `js-mobile-carousel` in `mobile-carousel.less`, so if the styles there were refactored it'd be easier to track down the dependent styles.

jimmynotjim replied:
Sorry, I meant you can write them on a single line since there's nothing else inside. Makes for a quicker scanning when you're trying to find matching selectors from the browser.

anselmbradford replied:
Ahh gotcha, fixed in https://github.com/cfpb/cfgov-refresh/commit/66bde16b50928cb33bc10a678254530d08487363

jimmynotjim commented on src/static/js/routes/the-bureau/bureau-structure/index.js:
```diff
@@ -8,22 +8,29 @@
+    return;
+  }
+
+  var bpBettings = {
```
Are these needed? They look like they're the same as the plugin defaults?

anselmbradford replied:
These are passed to the `BreakpointHandler` constructor, which doesn't have defaults. This is something that may get refactored away when updating the Slick carousel source b/c Slick now has responsive settings built in.

jimmynotjim replied:
Sorry, I'm confused, what makes this instance of the carousel different from the one on Working at the CFPB? Shouldn't they be able to be initialized the same way (including responsiveness?). Is that just outside the scope of this PR?

anselmbradford replied:
No all good. I was confused. Yeah they share the same code. Updated in https://github.com/cfpb/cfgov-refresh/commit/5ee8753e588140c3eeb0298bdc4983182f806b9f

Scotchester replied:
"Bettings"?

anselmbradford replied:
Typo fixed in https://github.com/cfpb/cfgov-refresh/commit/c3cd0a7555ed476357d10acb073619fa0ab60cb6 

As to the original comment. I was confused between which page this was referencing. `/the-bureau/` init code has been updated, while the `/the-bureau/bureau-structure/` uses the ContentSlider, not the MobileCarousel, so an update there is out of scope.

jimmynotjim commented on src/static/js/routes/careers/working-at-cfpb/index.js:
```diff
@@ -0,0 +1,21 @@
+/* ==========================================================================
```
Why do page named initializing instead of just looking for the `.js-mobile-carousel` class and using the same init on all the pages that include it (it doesn't seem they're any different)?

anselmbradford replied:
Because when per-page JS is implemented we don't want to be loading the Slick carousel code on every page. If we just have a few pages that have carousels they will have the same boilerplate initialization on the page-specific scripts, which I think is fine. If we start getting an unwieldy number of carousels we can move that initialization over to common.js to load it on all pages.

anselmbradford replied:
I did notice the-bureau needs the same initialization config though. Fixed in https://github.com/cfpb/cfgov-refresh/commit/5ee8753e588140c3eeb0298bdc4983182f806b9f

jimmynotjim replied:
I wasn't saying initialize everywhere, but it seems counter-intuitive to create duplicate inits on multiple index.js files when the same one could be used and just pulled into each page individually.

jimmynotjim commented:
Should the arrows be vertically centered on the image? They seem really odd way up high.

![screen shot 2015-06-29 at 6 40 20 pm](https://cloud.githubusercontent.com/assets/1280430/8420236/5a5ed0ce-1e8e-11e5-8169-5a7fb2324346.png)

anselmbradford commented:
> Should the dividers and the content footer align?

Yes, probably should. I'll see if I can track down what layout classes accomplish this.

> Should these align on the left?

This is due to the comment below. Not sure which to fix here yet.

> Looks like .content-l_col__before-inset-divider.content-l_col-1-2 has a border left of 30px all the time.

This is matching existing behavior of `.content-l_col__before-divider`, see [here](https://github.com/cfpb/cf-layout/blob/gh-pages/src/cf-layout.less#L567) and [here](https://github.com/cfpb/cf-layout/blob/gh-pages/src/cf-layout.less#L1449)

> Should the arrows be vertically centered on the image? They seem really odd way up high.

Yes, this is an oversight.

anselmbradford commented:
> Should the background image center as you change screen size?

Since this is a placeholder for a carousel, I'm going to not add styles till we're certain they'll be applicable to a carousel of images (likely added through slick carousel).

jimmynotjim commented:
> Since this is a placeholder for a carousel, I'm going to not add styles till we're certain they'll be applicable to a carousel of images (likely added through slick carousel).

Wait, this page is going to have two carousels? That seems a bit excessive. Also see http://shouldiuseacarousel.com/

anselmbradford commented:
Carousel arrow position adjusted in https://github.com/cfpb/cfgov-refresh/commit/bc8be0e92b4f14f2fb08e84a1a661dda539937ef

anselmbradford commented:
Added https://github.com/cfpb/cfgov-refresh/commit/f7cc6646c67db6df578086f32d90fc39934ac381 to fix issue where two carousels were being instantiated and colliding and breaking the carousel. Fix checks for page-specific classes on careers and the bureau.

anselmbradford commented:
> Should the dividers and the content footer align?

Discovered this is a site-wide issue. Out of scope with this PR, we need to confirm with design on the footer alignment. See /the-bureau/, /budget/, etc.

anselmbradford commented:
> Should these align on the left?

Discovered the mocks have this misalignment as well. Out of scope with this PR as this is a broader design/CF issue.
https://github.cfpb.gov/github-enterprise-assets/0000/0087/0001/1919/4acd91ba-168a-11e5-8843-95583507a3da.png

anselmbradford commented:
Okay, all comments addressed I think. Can I get another :+1: ?

anselmbradford commented:
Linkedin case copy still needs resolution, but we might follow that up in another PR since there are more careers pages to come yet that can adjust that as necessary.

jimmynotjim commented on src/static/css/mobile-carousel.less:
```diff
@@ -0,0 +1,103 @@
+        margin-top: unit(60px / @base-font-size-px, em);
+    });
+
+    .image-container {
```
This isn't very BEM. Instead of nesting these, they should probably have a `__mobile-carousel` modifier.

anselmbradford replied:
This one probably shouldn't be aware of the mobile-carousel. I looked at some other CF patterns, how about a pattern similar to `content`:

```
.content
    .content_hero
    .content_bar
    .content_wrapper
        .content_sidebar
        .content_main
```

So I'd update media to use:
```
.media
    .media_wrapper
        .media_image-container
            .media_image
        .media_body
```

…or…

```
.media
    .media_image-container
        .media_image
    .media_body
```

anselmbradford replied:
Okay, refactored this in https://github.com/cfpb/cfgov-refresh/commit/0c2e3ed9967b5a4d8b07c98f9e1a92fdf20d94dc and added a `media-stack` BEM block for defining vertically stacked media objects.

jimmynotjim commented on src/static/css/pages/bureau.less:
```diff
@@ -6,33 +6,21 @@
+// Override general slick carousel overrides defined in mobile-carousel.less.
+// Removes bottom border, because the-bureau has a bordered block below
+// its carousel.
+.the-bureau .js-mobile-carousel.slick-slider {
```
Even if you removed the js prefix, wouldn't `.the-bureau .slick-slider` suffice? We should really reduce specificity as much as we can.

anselmbradford replied:
The Slick carousel _could_ be used on the bureau page outside of the .js-mobile-carousel class. We shouldn't do that, but this protects it from breaking were someone (_not on this team, of course_) to do that.

jimmynotjim commented:
I think this might be a record setter, 104 comments. I'm sure @willbarton is going crazy, so to avoid driving him over the edge

SHIP IT
![4f5f88ed9e070-1](https://cloud.githubusercontent.com/assets/1280430/8464531/c3adb792-2010-11e5-9190-f0f12e567e16.gif)

anselmbradford commented:
yiss My favorite ship gif :smiley_cat:
